src/python_routing_v02/compute_nhd_routing_SingleSeg_v02.py

```diff
@@ -815,6 +865,43 @@ def main():
     if showtiming:
         print("... in %s seconds." % (time.time() - start_time))
 
+    # STEP 6
+    if showtiming:
+        start_time = time.time()
+    if verbose:
+        print("creating usgs time_slice data array ...")
+
+    if data_assimilation_parameters:
+        usgs_timeslices_folder = os.path.join(
+            root, "test/input/geo/nudgingTimeSliceObs/",
+        )
+        # routelink_subset_folder = os.path.join(
+        #     root,
+        #     "test/input/geo/routelink/",
+        # )
+        # "../../test/input/geo/routelink/routeLink_subset.nc"
+        # routelink_subset_file = routelink_subset_folder+"routeLink_subset.nc"
+        # data_assimilation_parameters["wrf_hydro_channel_ID_routelink_file"] = routelink_subset_file
+        # usgs_file_pattern_filter = "*.usgsTimeSlice.ncdf"
+
+        # usgs_files = glob.glob(usgs_timeslices_folder + usgs_file_pattern_filter)
+        # file_name = "2020-03-19_18:00:00.15min.usgsTimeSlice.ncdf"
+
+        usgs_df = nhd_io.get_usgs_from_time_slices(
+            data_assimilation_parameters["data_assimilation_parameters_file"],
+            usgs_timeslices_folder,
+            data_assimilation_parameters["data_assimilation_filter"],
+        )
+        # print(usgs_df)
+    else:
+        usgs_df = pd.DataFrame()
+    # da = nnu.build_channel_initial_state(data_assimilation_parameters["wrf_hydro_channel_ID_routelink_file"], usgs_df.index)
```

Comment:
- [ ] Before final commit, create the `nnu.build_data_assimilation` helper function and put all of 'Step 6' in there to make this more compact.